lib/mysql2/client.rb

```diff
@@ -47,6 +47,11 @@ def initialize(opts = {})
 
       ssl_options = opts.values_at(:sslkey, :sslcert, :sslca, :sslcapath, :sslcipher)
       ssl_set(*ssl_options) if ssl_options.any?
+      if opts[:ssl_mode]
+        m = parse_ssl_mode( opts[:ssl_mode] )
+        self.ssl_mode = m if m
+      end
```

Comment:
Works for me, we might end up passing nil values into ssl_mode=, and getting an error, but thats fine.